### PR TITLE
Add velocity control mode

### PR DESCRIPTION
**Pull request by dogoepp**
Here a new control mode is supported : sending commands to servos in velocity command mode (wheel in Robotis' jargon).

The mode is detected in the servos but we require that the user declares the desired control mode with a ros parameter, to be sure to detect wrong assumption on the current control mode of a servo.

### CODE REVIEW

**Comment by costashatz** [comment]
> Here a new control mode is supported : sending commands to servos in velocity command mode (wheel in Robotis' jargon).

Thanks for this nice PR..

> the user declares the desired control mode with a ros parameter

Is this really necessary? Or you did it for safety?

**Comment by dogoepp** [comment]
For the sole sake of safety I added this limitation to user-friendliness. I still wonder if it is a good idea.

**Comment by costashatz** [comment]
This overall seems nice! Only one question/remark: `ROS_ERROR_STREAM` and `ROS_FATAL_STREAM` do not exit the program right? I think we should terminate the program, especially for the `ROS_FATAL_STREAM` cases..

**Review by costashatz — COMMENTED** [review]
This overall seems nice! Only one question/remark: ROS_ERROR_STREAM and ROS_FATAL_STREAM do not exit the program right? I think we should terminate the program, especially for the ROS_FATAL_STREAM cases..

**Comment by dogoepp** [comment]
You're right for the fatal that should exit.

For the errors related to command interface type (position/velocity), the joints concerned are disabled. Is that not enough ? Do we need to give such a strong signal as to crash the whole program ?

File: src/dynamixel_control_hw.cpp [hunk 1]
```diff
@@ -62,55 +71,94 @@ int main(int argc, char** argv)
...
+                ROS_ERROR_STREAM("A command interface (speed or position) "
+                    << "should be declared for the actuator " << it->first
+                    << " or a default one should be defined with the parameter "
+                    << "'default_command_interface'.");
+            }
+        }
+    }
+    catch (XmlRpc::XmlRpcException& e) {
+        ROS_FATAL_STREAM("Exception raised when parsing the configuration: " << e.getMessage());
+        return 1;
```
Comment (costashatz) [inline src/dynamixel_control_hw.cpp]: Most probably naive question, but just making sure. This should be enough to not allow incorrect configurations right? I.e., nothing is needed in the `dynamixel_hardware_interface.hpp` file. Just making sure..

File: include/dynamixel_control_hw/dynamixel_hardware_interface.hpp [hunk 2]
```diff
@@ -212,47 +180,61 @@ namespace dynamixel {
...
                     }
+                    else if (OperatingMode::unknown != hardware_mode)
+                        ROS_ERROR_STREAM("Servo " << id << " was not initialised "
+                                                  << "(operating mode "
+                                                  << mode2str(hardware_mode)
+                                                  << " is not supported)");
+
+                    // enable torque output on the servo and set its configuration
+                    // including max speed
+                    _enable_and_configure_servo(_servos[i], hardware_mode);
```
Comment (costashatz) [inline include/dynamixel_control_hw/dynamixel_hardware_interface.hpp]: If I am not mistaken, when the `hardware_mode` is `OperatingModel::unknown` we should not enable the servo but from what I see we still `set_torque_enable(1)` in this function (`_enable_and_configure_servo`) no matter the mode. Of course, I might be missing something..

**Review by costashatz — COMMENTED** [review]
I like how this code is becoming. Good job @dogoepp! Some very minor comments...

File: src/dynamixel_control_hw.cpp (hunk 1 above)
Comment (dogoepp) [reply src/dynamixel_control_hw.cpp]: What is an incorrect configuration for you ?

The above code should only accept well-structures configuration. However, it does not prevent the addition of unused parameters (which are then ignored).
If a parameter is expected to be an integer but is of an other type, an exception is raised and causes the program to stop. Saddly, the exception message is not informative. I don't know how to improve this, except by splitting this try/catch in more, smaller ones.

Comment (costashatz) [reply src/dynamixel_control_hw.cpp]: > The above code should only accept well-structures configuration.

That's what I correct configuration is for me..

> If a parameter is expected to be an integer but is of an other type, an exception is raised and causes the program to stop.

And that's what an incorrect configuration is for me. :smile: 

> However, it does not prevent the addition of unused parameters (which are then ignored).

This is ok.

> Saddly, the exception message is not informative. I don't know how to improve this, except by splitting this try/catch in more, smaller ones.

I think we should not bother too much for this.. It is not of that big importance..

Great! This is ok for me then!

File: include/dynamixel_control_hw/dynamixel_hardware_interface.hpp (hunk 2 above)
Comment (dogoepp) [reply include/dynamixel_control_hw/dynamixel_hardware_interface.hpp]: You're right. Fixed

**Review by costashatz — APPROVED** [review]
@dogoepp this should be good to merge! Great job! Can you only check my last commit 
864bf9f ?

**Comment by dogoepp** [comment]
Looks good. Thanks !

**Merging**

**Pull request merged**